tests/publiccloud/check_registercloudguest.pm

```diff
@@ -34,9 +34,9 @@ sub run {
 
     if (get_var('PUBLIC_CLOUD_QAM', 0)) {
         $instance = $self->{my_instance} = $args->{my_instance};
-        $provider = $self->{provider} = $args->{my_provider};    # required for cleanup
+        $provider = $self->{my_provider} = $args->{my_provider};    # required for cleanup
```

Comment:
if you grep the code you will find pretty a lot of `{provider}` variables all over the project .  We need to discuss this ...